### ISSUE TITLE
Tests failing

### ISSUE THREAD

**Issue by chintak** [dnouri/nolearn#210]
I installed nolearn in a clean virtualenv on my machine using the install instructions provided. I tried running `py.test` as indicated in the _.travis.yml_ file, but it gave the following errror:
```
py.test: error: unrecognized arguments: --cov=nolearn --cov-report=term-missing
```
After commenting the corresponding line from _setup.cfg_, I was able to run `py.test`. This resulted in 5 test failures, mainly relating to Assertion errors for the MNIST Classification and `NaN` or very-large values being observed for Regression. The complete error log is attached in a file [err.txt](https://github.com/dnouri/nolearn/files/117209/err.txt) or viewed online here - [pastebin](http://pastebin.com/sGfJK2q0). 

I tried upgrading my `theano` and `lasagne` versions to the latest and also did the entire installation from scratch in another clean virtualenv to no luck. 

System details:
- GeForce GTX 960 GPU
- NVIDIA-SMI 352.63
- Driver Version: 352.63
- CUDA v7.0
- CuDNN v3.0

I was originally trying to use nolearn for a regression problem and I noticed similar `NaN` values from the very first _iteration_ (not even _epoch_) in spite of varying batch size, varying learning rate from 0.1 to 0.0001, using a simple 1-conv-2-dense layer architechture, and using `lasagne.init.Orthogonal` for initialization. All these combinations led to the same `NaN` problem.

Additionally, I have access to another system having a _GeForce GTX 650_ also running the same Driver and CUDA versions, where following the same installation instructions I am able to install nolearn successfully, it passes all the tests and I get meaningful train and valid losses for my regression problem using the same architecture I tried on 960.

Anything I might be missing?

**Comment by chintak** [comment]
Solved! It was a Theano related issue [#3953](https://github.com/Theano/Theano/issues/3953).
